pkg/collector/corechecks/snmp/value.go

```diff
@@ -0,0 +1,43 @@
+package snmp
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/metrics"
+	"strconv"
+)
+
+type snmpValueType struct {
+	submissionType metrics.MetricType // used when sending the metric
+	value          interface{}        // might be a `string` or `float64` type
+}
+
+func (sv *snmpValueType) toFloat64() float64 {
+	var retValue float64
+
+	switch sv.value.(type) {
+	case float64:
+		retValue = sv.value.(float64)
+	case string:
+		val, err := strconv.ParseInt(sv.value.(string), 10, 64)
+		if err != nil {
+			return float64(0)
+		}
+		retValue = float64(val)
+	}
+	// only float64/string are expected for now
```

Comment:
don't you need a default case throwing an error?

Reply:
It's not supposed to happen, so that will add some unused code. Not really necessary IMHO.

Reply:
It's probably the `for now` that made we wonder what would happen if one day we added another option

Reply:
As I remember having it on C++, found a linter option that warns for un-exhaustive switches: https://github.com/nishanths/exhaustive
If `sv.value.(type)` is typed to be only `float64|string` than this is ok, elsewhere IMO there should be a default catching possible errors.

Reply:
Yes, it's only `float64|string`, if we do have more types in the future, we can still improve the error handling :)

Reply:
Is it there a type guard lock somewhere? I personally prefer to have the system somehow telling me when boundaries change vs me reminding where I should update code

Reply:
@pducolin I changed the implementation to return en error, if any: https://github.com/DataDog/datadog-agent/pull/6749/files#diff-a835584b984e447b9b3ed420add42502491f312cd32ce8b25cbd318a209cd4fcR13-R35

The errors are handled by caller and tested.